backends/build_system/validate_env.py

```diff
@@ -30,12 +31,16 @@
 
 class UnmetDependenciesException(Exception):
     def __init__(self, unmet_deps):
+        pip_install_command = f"{sys.executable} -m pip install "
         msg = "Environment requires following Python dependencies:\n\n"
         for package, actual_version, required in unmet_deps:
             msg += (
                 f"{package} (required: {required.constraints}) "
                 f"(version installed: {actual_version})\n"
             )
+            pip_install_command += f'"{package}{required.string_constraints()}" '
```

Comment:
Instead of manually quoting these values, it may make sense to just do `shlex.join()` so that all arguments are properly escaped such as the python executable.

Reply:
Its the same net result.